### ISSUE TITLE
Can't install this on Mac OS Ventura...

### ISSUE THREAD

**Issue by tabascoman77** [section83/MacYTDL#67]
It attempts to drop stuff in my Library folders...but then kicks back a huge error and quits. After that, the app is completely inoperable with the same error.
![Screenshot 2023-05-16 at 3 08 27 PM](https://github.com/section83/MacYTDL/assets/87635043/2ad4d824-0680-455f-89ae-ab166d0f9a58)

**Comment by section83** [comment]
Many thanks for posting.

That error rings vague bells. MacYTDL doesn’t call anything in that location. So, something else is going on. Can you answer a few questions ?

- Are you using Homebrew or Macports installs of YT-DLP or FFmpeg ?
-  Have you installed Python - if so, which package ?
- From which location are you running MacYTDL ? Applications ?

I will have a closer look later today (Sydney time).

Cheers.

**Comment by tabascoman77** [comment]
1) I believe this is Homebrew.
2) I believe Python is installed but I don't recall which package.
3) Yes, the Apps folder.

**Comment by section83** [comment]
Thanks for that. I suspect MacYTDL is misbehaving on Macs which have a separate Python install. BTW, YT-DLP has a Python 3.8 runtime built in.

I can’t look at MacYTDL for at least a few days (family emergency). Will post here when I have something useful.

Thanks.

**Comment by section83** [comment]
[tabascoman77](https://github.com/tabascoman77)

I'm back from the family emergency and looking at this issue.  Have you found a fix for your problem ?

The "xcrun" error normally is not related to Python installs but is related to Xcode installs when a new version of macOS is installed. There are quite a lot of discussions about it although all posters are developers running into trouble with GIT and other developer tools. Theses are some of the discussions:

- [Xcode poblem](https://apple.stackexchange.com/questions/254380/why-am-i-getting-an-invalid-active-developer-path-when-attempting-to-use-git-a)
- [xcrun: error: invalid active developer path](https://developer.apple.com/forums/thread/673827)
- [[SOLVED] Mac OS: xcrun: error: invalid active developer path, missing xcrun](https://www.studytonight.com/post/solved-mac-os-xcrun-error-invalid-active-developer-path-missing-xcrun)

The fix nearly always involves this Terminal command:

`xcode-select --install`

Have you installed Xcode ?

**Comment by Infinivibex** [comment]
Does Xcode CLI tools work? If so, it could be helpful if it's stated in the install.

**Comment by section83** [comment]
Xcode and the CLI tools are **not** used/required for MacYTDL.

I'm not expert at all with Xcode and CLI tool.  However, I believe they can get out of sync, which might cause problems with facilities such as Python.  I think that if you are not a developer you do not need Xcode and CLI tools.  If you wish to retain Xcode and the separate Python and YT-DLP installs, I recommend looking at the discussions on xcrun error that I posted above.

I got the xcrun error in a app I use for notarizing MacYTDL.  The problem was a setting in Xcode. These images show the setting before the fix and after:

![Wrong Setting](https://github.com/section83/MacYTDL/assets/7471595/e33e9411-5ffd-4ab4-8c93-ea6f1af88a93)
![Setting Fixed](https://github.com/section83/MacYTDL/assets/7471595/805f954b-d5b4-4e32-acb9-46b6c61978af)
